src/extra/others/snapshot/lv_snapshot.c

```diff
@@ -148,6 +148,7 @@ lv_res_t lv_snapshot_take_to_buf(lv_obj_t * obj, lv_img_cf_t cf, lv_img_dsc_t *
     lv_mem_free(draw_ctx);
 
     dsc->data = buf;
+    dsc->data_size = buf_size_needed;
```

Comment:
I think this should be also `buf_size`

Reply:
`buf_size` is the available buffer size as passed into the function. But only `buf_size_needed` bytes will actually have valid image data.

Reply:
I agree to use `buf_size_needed` to zero only the really required buffer.

Reply:
Sorry, I missed something. So the questions is if need to use the buf size passed by the user (the user might want to know the size he was using) or set it for the image (LVGL might want to know the real buffer size of image).

I vote for 2) so `dsc->data_size = buf_size_needed;`.

I merge this PR now, if you think that it would be important to discuss it, please open an issue.